### ISSUE TITLE
Don't allow `@final` methods to be reassigned

### ISSUE THREAD

**Issue by saaketp** [microsoft/pyright#3448]
**Is your feature request related to a problem? Please describe.**
I want to prevent a method from being reassigned, for attributes I'd use `typing.Final` for that. But I don't see any simple way to say that a method is `Final`. I thought the `@final` decorator would do it, but pyright allows reassigning `@final` methods.

For example, this passed pyright strict mode without errors:
```python
from typing import final

class A:
    @final
    def f(self) -> None:
        pass

A.f = lambda self: None
```

**Describe the solution you'd like**
I'd like to see an error similar to `Final` annotated attributes, like `Cannot assign member "f" for type "Type[A]"
  "f" is declared as Final and cannot be reassigned.`

**Additional context**
Mypy treats `@final` as not reassignable and gives error for the above code `Cannot assign to final attribute "f"  [misc]`.
Weirdly, it also just doesn't allow assigning to methods at all, regardless of `@final` so it gives another error `Cannot assign to a method  [assignment]`

**Comment by erictraut** [comment]
Thanks for the feature request.

[PEP 591](https://peps.python.org/pep-0591/), which introduced `@final` doesn't specifically indicate that `@final` methods cannot be reassigned, but it seems like this is keeping with the spirit of the intent. The fact that mypy enforces lends further support for this interpretation.

I'll work on adding this support shortly.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.246, which I just published. It will also be included in the next release of pylance.
